### PR TITLE
Plane: support send_extended_sys_state

### PR DESCRIPTION
**Pull request by IamPete1**
Rework of https://github.com/ArduPilot/ardupilot/pull/10698 now we have separate translation logic.

### CODE REVIEW

**Comment by IamPete1** [comment]
Seems to work, tested both SLT and tailsitters using:
```
long 511 245 100000 
graph EXTENDED_SYS_STATE.vtol_state
```

File: ArduPlane/quadplane.cpp [hunk 1]
```diff
@@ -3687,4 +3687,28 @@ bool SLT_Transition::active() const
     return quadplane.assisted_flight && ((transition_state == TRANSITION_AIRSPEED_WAIT) || (transition_state == TRANSITION_TIMER));
 }
 
+MAV_VTOL_STATE SLT_Transition::get_mav_vtol_state() const
+{
+    if (quadplane.in_vtol_mode()) {
+        QuadPlane::position_control_state state = quadplane.poscontrol.get_state();
+        if ((state == QuadPlane::position_control_state::QPOS_AIRBRAKE) || (state == QuadPlane::position_control_state::QPOS_POSITION1)) {
```
Comment (IamPete1) [inline ArduPlane/quadplane.cpp:3694]: might want to limit this to only be `QPOS_AIRBRAKE` we can sometimes spend a while repositioning in `QPOS_POSITION1`

**Review by peterbarker — APPROVED** [review]
*So* much nicer than my effort :-)

File: ArduPlane/GCS_Mavlink.cpp [hunk 2]
```diff
@@ -1417,3 +1417,27 @@ int8_t GCS_MAVLINK_Plane::high_latency_air_temperature() const
...
+#else
+    if (!plane.quadplane.available()) {
+        return MAV_VTOL_STATE_UNDEFINED;
+    }
+
+    return plane.quadplane.transition->get_mav_vtol_state();
+#endif
+};
+
+MAV_LANDED_STATE GCS_MAVLINK_Plane::landed_state() const
```
Comment (khancyr) [inline ArduPlane/GCS_Mavlink.cpp:1434]: we could work the takeoff and landing state later

**Review by khancyr — APPROVED** [review]
It would be nice to have the autotest too!

**Comment by IamPete1** [comment]
The original PR from @peterbarker did have some auto tests. I guess they should still work.....

**Comment by khancyr** [comment]
yep, that is why I asked ! specially when the vtol transition seems tricky

**Review by tridge — APPROVED** [review]

**Pull request merged**